### PR TITLE
New parsing docs type definition flow

### PR DESCRIPTION
**Pull request by omrimn**

### CODE REVIEW

File: docworks-dts/lib/dts-generator.js [hunk 1]
```diff
@@ -49,6 +54,61 @@ function getDtsType(type, options = {}) {
...
+  return type.some(currentType => currentType.nativeType || currentType.complexType || currentType.referenceType)
+}
+
+function getDtsFromDocsType(docsTypes) {
+  if(isUnionType(docsTypes)){
+    return docsTypes.map(t => getDtsFromDocsType([t])).join(' | ')
+  }
+  else{
+    const docsType = docsTypes[0]
+    if(docsType.nativeType || docsType.referenceType){
```
Comment (nemni8) [inline docworks-dts/lib/dts-generator.js:68]: it's strange to have a recursive that doesn't start with the stop condition.
imo the function should start with `if(docsType.nativeType || docsType.referenceType)` condition and all other recursive calls should be in the else block

**Review by nemni8 — COMMENTED** [review]
looks good! 
had a few comments

File: docworks-dts/lib/dts-generator.js [hunk 2]
```diff
@@ -49,6 +54,61 @@ function getDtsType(type, options = {}) {
...
+}
+
+// Array of more one value defines for us a union type.
+function isUnionType(docsTypes) {
+  return docsTypes.length > 1
+}
+
+// Supported key value types: Record/Map.
+function isKeyValueType(typeName) {
+  return typeName === 'Record' || typeName === 'Map'
```
[the remark this answers is not in the document]
Comment (omrimn) [reply docworks-dts/lib/dts-generator.js:91]: Removed this, if we will want we will do it in metro side

File: docworks-dts/lib/dts-converters.js [hunk 3]
```diff
@@ -6,11 +6,14 @@ const {
   dtsInterface,
   dtsMethod,
   dtsModule,
+  dtsEnum,
   dtsParameter,
   dtsProperty,
   dtsObjectTypeAlias
 } = require('./dts-generator')
 
+const getServiceSummary = service => service && service.docs && service.docs.summary ? service.docs.summary : ''
```
[the remark this answers is not in the document]
Comment (omrimn) [reply docworks-dts/lib/dts-converters.js:15]: Yes but it seems there's an issue there and we want to move forward with this so I prefer that at the moment we will protect ourselves with this.

File: docworks-dts/lib/dts-generator.js [hunk 4]
```diff
@@ -49,6 +54,61 @@ function getDtsType(type, options = {}) {
...
+    if(typeParamA.key === true){
+      return -1
+    }
+    else if(typeParamB.key === true){
+      return 1
+    }
+    return 0
+  })
+
+  const value = typeParamsByKeyIfExists.splice(1)
```
[the remark this answers is not in the document]
Comment (omrimn) [reply docworks-dts/lib/dts-generator.js:106]: I like the multiline more clear to me.
The macro is nice to reduce lines but it takes a second longer to understand it. 
I don't mind taking it. 
And the destruct to array doesnt work.

[the remark this answers is not in the document]
Comment (nemni8) [reply docworks-dts/lib/dts-generator.js:106]: it's your call.
WDYM the destruct doesn't work..? if the sort returns an array with 2 items then it should work..

File: docworks-dts/lib/dts-generator.js (hunk 1 above)
Comment (nemni8) [reply docworks-dts/lib/dts-generator.js:68]: i prefer starting with stop condition, but it's not critical

**Review by nemni8 — APPROVED** [review]

**Pull request merged**